src/components/NewEventButton.js

```diff
@@ -19,7 +20,24 @@ const NewEventScreen = () => {
   const [dateTimePicker, setDateTimePicker] = useState(false)
   const [placeName, setPlaceName] = useState('')
   const [date, setDate] = useState(new Date())
-  const [longitude, latitude] = useSelector(selectedCoordinatesSelector)
+  const [latitude, setLatitude] = useState()
+  const [longitude, setLongitude] = useState()
+  const dispatch = useDispatch()
+
+  const selectedCoordinates = useSelector(selectedCoordinatesSelector)
+  const currentPosition = useSelector(currentPositionSelector)
+
+  useEffect(() => {
+    if (selectedCoordinates) {
+      setLatitude(selectedCoordinates.latitude)
+      setLongitude(selectedCoordinates.longitude)
+    } else {
+      if (currentPosition) {
+        setLatitude(currentPosition.latitude)
+        setLongitude(currentPosition.longitude)
+      }
+    }
+  }, [currentPosition, selectedCoordinates])
```

Comment:
This shouldn't be in two `useEffect` functions?

Reply:
i would rewrite it in this way:
```
useEffect(() => {
  if (!selectedCoordinates) return
  set...
},[selectedCoordinates])

useEffect(() => {
  if (selectedCoordinates) return
  if (!currentPosition) return
  set...
}, [currentPosition])
```

But maybe it is wrong, and maybe it has no effect on readability.